src/main/java/com/aws/greengrass/tes/CredentialRequestHandler.java

```diff
@@ -115,16 +126,29 @@ public CredentialRequestHandler(final IotCloudHelper cloudHelper, final IotConne
 
         deviceConfiguration.getIotRoleAlias().subscribe((why, newv) -> {
             String iotRoleAlias = Coerce.toString(newv);
-            clearCache();
-            setIotCredentialsPath(iotRoleAlias);
+            if (Utils.isNotEmpty(iotRoleAlias)) {
+                clearCache();
+                setIotCredentialsPath(iotRoleAlias);
+            }
```

Comment:
I don't think this is quite right - `useTesV1` needs to be definitive otherwise you get into situations where the order in which config is updated or read changes the behavior, right? Role alias should only be processed if `useTesV1` is absent or false.

Example:
* Customer configures `useTesV1 == true` -> TES switches to V1
* Customer updates `iotRoleAlias` -> TES switches back to V2
* Customer restarts Nucleus -> TES processes `useTesV1` last and uses V1

Reply:
It may be a stop gap solution, but once we ship it, we can't remove it.

I can see this being an issue after a customer migrates from V1 back to V2. From an ops perspective, we need to be able to know what GG is doing based on just the config. Should be simple to fix, no?

Reply:
Updated